File: src/ch04/code_2.rs
fn recursive_func(n: i32) -> i32 {
    println!("func({})を呼び出しました", n);
    if n == 0 {
        return 0;
    }
    let result: i32 = n + recursive_func(n-1);
    println!("{} までの和 = {}", n, result);
    result
}


fn main() {
    recursive_func(5);
}

File: src/ch04/code_2.py
from rich import print


def recursive_func(n: int) -> int:
    print(f'func({n})を呼び出しました')
    if n == 0:
        return 0
    result = n + recursive_func(n=n-1)
    print(f'{n} までの和 = {result}')
    return result


if __name__ == '__main__':
    recursive_func(n=5)